src/Shared/HttpSys/Extensions.cs

```diff
@@ -2,16 +2,17 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using Microsoft.AspNetCore.Http;
 
 namespace Microsoft.AspNetCore.HttpSys.Internal
 {
     internal static class Extensions
     {
         public static string GetHttpProtocolVersion(this Version version) => version switch
```

Comment:
Could we replace this API with one on HttpProtocols? And its inverse API (string -> version)?

Reply:
This is used by HttpSys to go from Version to string for protocol. It needs to be in this direction.

Reply:
I know, I meant HttpProtocols should have APIs to convert both to and from a Version.

Reply:
I have removed GetHttpProtocolVersion and added Version -> string.

If someone needs the other direction then they can add it later.